ste/jobStatusManager.go

```diff
@@ -61,15 +61,34 @@ func (jm *jobMgr) statusMgrClosed() bool {
 
 /* These functions should not fail */
 func (jm *jobMgr) SendJobPartCreatedMsg(msg JobPartCreatedMsg) {
-	jm.jstm.partCreated <- msg
-	if msg.IsFinalPart {
-		// Inform statusManager that this is all parts we've
-		close(jm.jstm.partCreated)
+	defer func() {
+		if recErr := recover(); recErr != nil {
+			jm.Log(common.LogError, "Cannot send message on closed channel")
+		}
+	}()
+	if jm.jstm.partCreated != nil { // Sends not allowed if channel is closed
```

Comment:
Do we need to do this check?

Reply:
Yes, because sending on a nil channel will cause a deadlock and block the routine indefinitely.

Reply:
If it is nil should an error be returned? Also should we add this nil check to the xferDoneMsg channel?

Reply:
The channel is set to nil only after it's been closed. And, we already have support for cleanly erroring out for sends on a closed channel.